### PR TITLE
[FIX] Run ICA-AROMA in ``MNI152Lin`` 2mm resampling grid (91x109x91 vox)

### LINKED ISSUE #870: [BUG] ICA-AROMA - Mask and image must be the same size
From https://neurostars.org/t/runtime-error-invalid-literal-for-int-with-base-10-mask-and-image-must-be-the-same-size-during-ica-aroma-fmriprep-1-0-0-rc8/985?u=oesteban

```
Node: fmriprep_wf.single_subject_105_wf.func_preproc_task_future_run_01_wf.bold_confounds_wf.ica_aroma_wf.ica_aroma
Working directory: /root/src/fmriprep/work/fmriprep_wf/single_subject_105_wf/func_preproc_task_future_run_01_wf/bold_confounds_wf/ica_aroma_wf/ica_aroma

Node inputs:

TR = <undefined>
args = <undefined>
compress_report = auto
denoise_type = nonaggr
dim = <undefined>
environ = {}
feat_dir = <undefined>
fnirt_warp_file = <undefined>
generate_report = True
ignore_exception = False
in_file = /root/src/fmriprep/work/fmriprep_wf/single_subject_105_wf/func_preproc_task_future_run_01_wf/bold_confounds_wf/ica_aroma_wf/smooth/vol0000_xform-00000_merged_smooth.nii.gz
mask = <undefined>
mat_file = <undefined>
melodic_dir = /root/src/fmriprep/work/fmriprep_wf/single_subject_105_wf/func_preproc_task_future_run_01_wf/bold_confounds_wf/ica_aroma_wf/melodic
motion_parameters = /root/src/fmriprep/work/fmriprep_wf/single_subject_105_wf/func_preproc_task_future_run_01_wf/bold_hmc_wf/normalize_motion/motion_params.txt
out_dir = <undefined>
out_report = ica_aroma_reportlet.svg
report_mask = /root/src/fmriprep/work/fmriprep_wf/single_subject_105_wf/func_preproc_task_future_run_01_wf/bold_mni_trans_wf/mask_mni_tfm/ref_image_corrected_trans_masked_corrected_brain_mask_maths_trans.nii.gz
terminal_output = <undefined>

Traceback (most recent call last):
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/plugins/multiproc.py", line 51, in run_node
    result['result'] = node.run(updatehash=updatehash)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/engine/nodes.py", line 407, in run
    self._run_interface()
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/engine/nodes.py", line 517, in _run_interface
    self._result = self._run_command(execute)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/engine/nodes.py", line 650, in _run_command
    result = self._interface.run()
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/interfaces/base.py", line 1088, in run
    runtime = self._run_interface(runtime)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/common/report.py", line 55, in _run_interface
    ReportCapableInterface, self)._run_interface(runtime)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/interfaces/base.py", line 1654, in _run_interface
    self.raise_exception(runtime)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/interfaces/base.py", line 1600, in raise_exception
    **runtime.dictcopy()))
RuntimeError: Command:
ICA_AROMA.py -den nonaggr -i /root/src/fmriprep/work/fmriprep_wf/single_subject_105_wf/func_preproc_task_future_run_01_wf/bold_confounds_wf/ica_aroma_wf/smooth/vol0000_xform-00000_merged_smooth.nii.gz -meldir /root/src/fmriprep/work/fmriprep_wf/single_subject_105_wf/func_preproc_task_future_run_01_wf/bold_confounds_wf/ica_aroma_wf/melodic -mc /root/src/fmriprep/work/fmriprep_wf/single_subject_105_wf/func_preproc_task_future_run_01_wf/bold_hmc_wf/normalize_motion/motion_params.txt -o /root/src/fmriprep/work/fmriprep_wf/single_subject_105_wf/func_preproc_task_future_run_01_wf/bold_confounds_wf/ica_aroma_wf/ica_aroma
Standard output:

------------------------------- RUNNING ICA-AROMA ------------------------------- 
--------------- 'ICA-based Automatic Removal Of Motion Artifacts' --------------- 

Step 1) MELODIC
  - The existing/specified MELODIC directory will be used.
  - The MELODIC directory does not contain the required 'stats' folder. Mixture modeling on the Z-statistical maps will be run.
Step 2) Automatic classification of the components
  - registering the spatial maps to MNI
  - extracting the CSF & Edge fraction features
Standard error:
Traceback (most recent call last):
  File "/opt/ICA-AROMA/ICA_AROMA.py", line 200, in <module>
    edgeFract, csfFract = aromafunc.feature_spatial(fslDir, outDir, scriptDir, melIC_MNI)
  File "/opt/ICA-AROMA/ICA_AROMA_functions.py", line 388, in feature_spatial
    '-V | awk \'{print $1}\''])))
ValueError: invalid literal for int() with base 10: 'Mask and image must be the same size'
Return code: 1
```

### PR DESCRIPTION
**Pull request by oesteban**
This PR is an attempt to finally close #870, by ensuring that ICA-AROMA is run in the MNI152Lin 2mm image grid (i.e. 91x109x91 voxels). This is opposed to the previous implementation which worked on the original grid of the image (after resampling into MNI space) or the 2mm MNI152 nonlinear asymmetric 2009c (i.e. 97x115x97) if the 2mm option was selected.

To do so, the corresponding workflow has been modified, including an internal BOLD resampling workflow. When the user specifies ``MNI152Lin`` as template space, this workflow will be duplicated. One possible optimization to consider in the future is identifying that situation, and avoid the outer resampling when AROMA is selected. However, for a first take I assumed this duplication is not too costly.

Other improvements included with this PR are:

  * [x] Update ICA-AROMA to the latest 0.4.4 release.
  * [x] Remove the check of ``--use-aroma`` and template.
  * [x] Update documentation accordingly.
  * [x] Update naming of non-aggresively denoised images to the new space

### CODE REVIEW

**Comment by oesteban** [comment]
It'd be great if you could have a look, @jdkent

**Comment by chrisfilo** [comment]
Do you have some example data you could use to check this modification solves the issue?

I also don't quite follow why why need to change the coregistration target (aka template) not just the resampling target (to get the right voxel size and sampling grid). Could you clarify?

**Comment by oesteban** [comment]
Oh, my bad, this only changes the resampling target. I'll fix those names.

File: fmriprep/workflows/bold/base.py [hunk 1]
```diff
@@ -603,45 +603,51 @@ def init_func_preproc_wf(bold_file, ignore, freesurfer,
...
+            (bold_sdc_wf, ica_aroma_wf, [
+                ('outputnode.out_warp', 'inputnode.fieldwarp')]),
+            (bold_confounds_wf, join, [
+                ('outputnode.confounds_file', 'in_file')]),
+            (ica_aroma_wf, join,
+                [('outputnode.aroma_confounds', 'join_file')]),
+            (ica_aroma_wf, outputnode,
+                [('outputnode.aroma_noise_ics', 'aroma_noise_ics'),
+                 ('outputnode.melodic_mix', 'melodic_mix'),
+                 ('outputnode.nonaggr_denoised_file', 'nonaggr_denoised_file')]),
```
Comment (jdkent) [inline fmriprep/workflows/bold/base.py]: If we eventually support denoising in multiple spaces, we may want to give the non-aggressively denoised file a more specific name, such as `nonaggr_denoised_mni`, but this change is not directly related to this pull request.

**Comment by oesteban** [comment]
Thanks a lot @jdkent, you are right in all the points you raised. I'll revise this PR.

**Comment by chrisfilo** [comment]
@jdkent did you test this PR on a dataset that was previously causing tthe AROMA pipeline to error?

**Comment by chrisfilo** [comment]
I'm seeing 
```
RuntimeError: Command:
ICA_AROMA.py -den nonaggr -i /scratch/fmriprep_wf/single_subject_01_wf/func_preproc_task_mixedgamblestask_run_01_wf/ica_aroma_wf/smooth/vol0000_xform-00000_merged_smooth.nii.gz -meldir /scratch/fmriprep_wf/single_subject_01_wf/func_preproc_task_mixedgamblestask_run_01_wf/ica_aroma_wf/melodic -mc /scratch/fmriprep_wf/single_subject_01_wf/func_preproc_task_mixedgamblestask_run_01_wf/bold_hmc_wf/normalize_motion/motion_params.txt -o /scratch/fmriprep_wf/single_subject_01_wf/func_preproc_task_mixedgamblestask_run_01_wf/ica_aroma_wf/ica_aroma
Standard output:

Standard error:
Segmentation fault (core dumped)
Return code: 139
```
in CI. 

External testers might wanna hold off until this is fixed.

**Comment by jdkent** [comment]
The `ica_aroma_wf` doesn't even exist in artifacts, does that mean it's not being created or not uploaded to artifacts, or unknown?

**Comment by oesteban** [comment]
I'm working on it :+1:

**Comment by oesteban** [comment]
@chrisfilo : 

Just filed https://github.com/maartenmennes/ICA-AROMA/issues/24

@jdkent :

> The `ica_aroma_wf` doesn't even exist in artifacts, does that mean it's not being created or not uploaded to artifacts, or unknown?

Neither in the master branch. For some reason we must be wiping those out at some point.

**Comment by chrisfilo** [comment]
To avoid confusion when someone looks at this PR in the changelog in the future, could we adjust the name and description to explicitly state that it's not the space (coregistration target), but sampling grid that is matched to one particular instance of MNI for AROMA?

**Comment by oesteban** [comment]
@chrisfilo better now?

**Comment by chrisfilo** [comment]
Fantastic - thanks.

**Comment by oesteban** [comment]
This is ready for review, @chrisfilo :+1:

**Comment by oesteban** [comment]
As per https://neurostars.org/t/fmriprep-error-when-performing-aroma-and-select-template-resampling-grid-2mm/1566/12?u=oesteban, this patch fixes the issue for that use case.
